packages/utils/src/validation/yup.js

```diff
@@ -126,11 +126,51 @@ const testSync = (schema, createError) =>
     },
   });
 
+/**
+ * Wrap a test function that throws an error upon failure for using in yup
+ * @param {(...args: any[]) => any} testFunction
+ * @param {string} [message]
+ * @returns
+ */
+const yupTest = (testFunction, message) => (...args) => {
```

Comment:
The motivation for adding this function is that `yup.test` wants to take a function that returns either a boolean or a `yup.ValidationError`.

We have a fairly comprehensive suite of validation checks in `utils/src/validation`. Unfortunately, these all throw an error if they fail, and they're designed to work with our `ObjectVadidator` class. That class is pretty heavily used still, so I didn't really want to touch anything. This is my workaround to be able to share that functionality.